src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -62,4 +55,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json");
     response.getWriter().println(messageJson);
   }
+
+  /**
+   * Post is not supported by GmailServlet.
+   *
+   * @param request HTTP request from client
+   * @param response Response which will contain 400 error
+   * @param googleCredential valid google credential object (already verified)
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doPost(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    response.sendError(400, "Post is not supported");
```

Comment:
Maybe throw [UnsupportedOperationException](https://docs.oracle.com/javase/6/docs/api/java/lang/UnsupportedOperationException.html) so as to fail fast? (I don't have a strong opinion in this case, it's just food for thought.)

Reply:
Alternatively, put this in the implementation for `doPost` in `AuthenticatedHttpServlet` (it won't be an abstract method any more, but that's ok).

That way, derived classes don't need to implement `doPost` if it is not supported.

I don't have a strong feeling about returning a 400 response vs. throwing an exception. The 400 response seems like a more accurate error. (I assume exceptions would result in a 500 response.)

Reply:
Throwing an exception would result in a 500 error, which I would prefer to reserve for actual implementation issues on the server side (this error would just be an illegal call). 

I will include this in the AuthenticatedHttpServlet however as a default implementation.